src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js

```diff
@@ -0,0 +1,104 @@
+import * as _ from 'lodash';
+import {
+  joinActivity,
+  startConversationActivity,
+} from '../../../actions/index';
+import piiReplace from '../piiReplace';
+import {
+  IN_AUTH_EXP,
+  IS_TRACKING_UTTERANCES,
+  LOGGED_IN_FLOW,
+  RECENT_UTTERANCES,
+} from '../../chatbox/utils';
+
+// define thunks for actions
+export const processActionConnectFulfilled = ({
+  requireAuth,
+  dispatch,
+  csrfToken,
+  apiSession,
+  apiURL,
+  baseURL,
+  userFirstName,
+  userUuid,
+}) => () => {
+  if (requireAuth && sessionStorage.getItem(LOGGED_IN_FLOW) !== 'true') {
+    dispatch(
+      startConversationActivity(
+        csrfToken,
```

Comment:
optional/minor: can wrap all args into an object and pass that object. Since there are quite a few args, this could be cleaner